src/xref2/ref_tools.ml

```diff
@@ -139,14 +139,25 @@ let find find sg name =
   | None -> Error (`Find_by_name (`Any, name))
 
 let module_lookup_to_signature_lookup env (ref, cp, m) =
-  Tools.signature_of_module env m
+  let rec handle_expansion : Tools.expansion -> _ = function
```

Comment:
We could have a function in `Expand_tools` to do that. eg. `handle_expansion_sig`.